Documentation/ContentObjects/UserAndUserInt/_ExampleListRecords.php

```diff
@@ -27,24 +36,28 @@ public function setContentObjectRenderer(ContentObjectRenderer $cObj): void
     /**
      * List the headers of the content elements on the page
      *
-     * @param  string Empty string (no content to process)
-     * @param  array  TypoScript configuration
+     * @param string Empty string (no content to process)
+     * @param array TypoScript configuration
+     * @param ServerRequestInterface The current PSR-7 request object
      * @return string HTML output, showing content elements (in reverse order, if configured)
      */
     public function listContentRecordsOnPage(string $content, array $conf, ServerRequestInterface $request): string
     {
-        $connection = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionForTable('tt_content');
+        $connection = $this->connectionPool->getConnectionForTable('tt_content');
         $result = $connection->select(
             ['header'],
             'tt_content',
-            ['pid' => (int)$GLOBALS['TSFE']->id],
+            // The request attribute 'frontend.page.information' has been introduced
```

Comment:
I would mention such information in the text not as a coment in the code

Reply:
This was by intention: Then it is to far away from the context. Ppl tend to just copy/paste without reading text above or below the code snippet. So, they copy the comment with it.

Reply:
If they copy this comment they start wondering what a version switch is as they are then not in the context of the docs anymore